### PR TITLE
fix: explorer not showing DG

### PR DESCRIPTION
**Pull request by thelostone-mc**
Note this can be tested only on V1 to make sure Uniswap DG round is displayed

https://discord.com/channels/562828676480237578/1239738214474842163/1239850037178273842

- Ensure DG grants shows up on the explorer
- Adds a new filter to ensure private rounds are displayed on the explorer

Screenshot 2024-05-14 at 3 26 01 PM

### CODE REVIEW

File: packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx [hunk 1]
```diff
@@ -18,14 +18,17 @@ const ExploreRoundsPage = () => {
   // Pass the filter from the search params and build the graphql query
   const rounds = useFilterRounds(filter, getEnabledChains());
 
+  const publicRounds = rounds.data?.filter(round => round.roundMetadata.roundType?.toLowerCase() !== "private");
```
Comment (boudra) [inline packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx:21]: this is potentially an expensive calculation that we're running on each render, (filtering up to 500 rounds) will slow things down for users considerably: https://outshift.cisco.com/blog/avoiding-expensive-recalculation-in-react-with-the-usememo-hook

File: packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx [hunk 2]
```diff
@@ -18,14 +18,17 @@ const ExploreRoundsPage = () => {
...
+  rounds.data = publicRounds;
```
Comment (boudra) [inline packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx:22]: this will surely blow up in the future in unexpected ways.

**Review by boudra — APPROVED** [review]
added some comments but approved, please do what you feel is best.

File: packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx (hunk 2 above)
Comment (thelostone-mc) [reply packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx:22]: Yup! We likely have make this as custom function on the indexer. Will make an issue to track and fix it up this week

Comment (thelostone-mc) [reply packages/grant-explorer/src/features/discovery/ExploreRoundsPage.tsx:22]: https://github.com/gitcoinco/grants-stack/issues/3433

**Pull request merged**